concordium-consensus/src/Concordium/GlobalState/Basic/BlockState.hs

```diff
@@ -2112,4 +2088,32 @@ genesisState gd = case protocolVersion @pv of
                         hashShouldMatch = case migration of
                             StateMigrationParametersTrivial{} -> True
                             StateMigrationParametersP3ToP4{} -> False
-                        genesisTT = runIdentity $ runPureBlockStateMonad $ BS.getInitialTransactionTable hbs
+                        genesisTT = getInitialTransactionTable hbs
+
+-- |Construct a transaction table that is empty but records the next nonces/sequence numbers
+-- for all accounts and chain updates.
```

Comment:
Isn't this comment somewhat misleading? The transaction table only explicitly records the nonces for accounts which have non-initial nonces.